src/components/Technologies/Technologies.tsx

```diff
@@ -13,68 +18,151 @@ const searchOptions = {
+    let _selectedFilters = [] as SearchFilter[];
+    Object.keys(selectedFilters).forEach((key) => {
+      const _values = selectedFilters[key as keyof typeof selectedFilters];
+      if (!_values || _values.length === 0) {
+        return;
       }
-
-      if (category1 > category2) {
-        return 1;
+      if (key === "additionalFilters") {
+        const additionalFilters = Array.isArray(_values) ? _values.reduce((acc, _value) => {
+          acc.push({
+            name: _value,
+            values: true
+          });
+          return acc;
+        }, [] as SearchFilter[]) : [];
+        _selectedFilters = _selectedFilters.concat(additionalFilters);
+      } else {
+        _selectedFilters.push({
+          name: key,
+          values: _values
+        });
       }
-
-      return 0;
     });
-
+    let filteredCards: any[] = [];
+    filteredCards = data.filter((techCard) => {
+      const card = techCard.fields;
+      return _selectedFilters.every((condition) => {
+        if (condition.name === "category") {
+          return condition.values.includes(card.packType);
+        } else if (condition.name === "registries") {
+          return condition.values.some((registry: string) => card.registries.includes(registry));
+        } else if (condition.name === "cloudTypes") {
+          return card.cloudTypes.includes("all") || card.cloudTypes.includes(condition.values);
+        } else if (["verified", "community"].includes(condition.name)) {
+          return card[condition.name as keyof FrontMatterData];
+        } else {
+          return true;
+        }
+      })
+    })
     if (searchValue) {
-      const fuse = new Fuse(technologies, searchOptions);
-      technologies = fuse.search(searchValue).map(({ item }) => item);
-    }
-
-    if (selectedCategory !== "all") {
-      technologies = technologies.filter(({ fields }) => fields.category.includes(selectedCategory)) || [];
+      const fuse = new Fuse(filteredCards, searchOptions);
+      filteredCards = fuse.search(searchValue).map(({ item }) => item);
     }
+    const categoriesMap = filteredCards.reduce((acc: Map<string, any>, technology: PacksData | IntegrationsData) => {
+      let packType = technology.fields.packType;
+      if (acc.has(packType)) {
+        acc.get(packType).push(technology.fields);
+      } else {
+        acc.set(packType, [technology.fields]);
+      }
+      return acc;
+    }, new Map<string, any>());
 
-    return technologies;
-  }, [data, searchValue, selectedCategory]);
+    const sortedCategoriesMap = new Map([...categoriesMap.entries()].sort((field1: string, field2: string) => {
+      const packType1: keyof typeof packTypeNames = field1;
+      const packType2: keyof typeof packTypeNames = field2;
+      if (packTypeNames[field1[0]] > packTypeNames[field2[0]]) {
+        return 1;
+      } else if (packTypeNames[field1[0]] < packTypeNames[field2[0]]) {
+        return -1;
+      } else {
+        return 0;
+      }
```

Comment:
not how I'd compare strings. 
note: all of the if-else can be cleaner
my suggestion is to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare